airflow-core/src/airflow/api_fastapi/core_api/routes/public/log.py

```diff
@@ -151,3 +151,48 @@ def get_log(
             "Airflow-Continuation-Token": URLSafeSerializer(request.app.state.secret_key).dumps(metadata)
         }
     return Response(media_type="application/x-ndjson", content=logs, headers=headers)
+
+
+@task_instances_log_router.get(
+    "/{task_id}/externalLogUrl/{try_number}",
+    responses={
+        **create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND]),
+        status.HTTP_200_OK: {
+            "description": "External log URL",
+            "content": {
+                "application/json": {"schema": {"type": "object", "properties": {"url": {"type": "string"}}}}
+            },
+        },
+    },
+    dependencies=[Depends(requires_access_dag("GET", DagAccessEntity.TASK_INSTANCE))],
+    response_model=ExternalLogUrlResponse,
+)
+def get_external_log_url(
+    dag_id: str,
+    dag_run_id: str,
+    task_id: str,
+    try_number: PositiveInt,
+    session: SessionDep,
+    map_index: int = -1,
+):
```

Comment:
Nit: The `get_log` endpoint that you take reference is a special one ( it will return `application/json` and `application/x-ndjson` based on accept header ). For most of the routes, we tend to add annotation instead of `response_model`.

```suggestion
@task_instances_log_router.get(
    "/{task_id}/externalLogUrl/{try_number}",
    responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND]),
    dependencies=[Depends(requires_access_dag("GET", DagAccessEntity.TASK_INSTANCE))],
)
def get_external_log_url(
    dag_id: str,
    dag_run_id: str,
    task_id: str,
    try_number: PositiveInt,
    session: SessionDep,
    map_index: int = -1,
) -> ExternalLogUrlResponse:
```

Reply:
I would keep this in mind, thanks for figuring out!